Library/include/CSP/Services/WebService.h

```diff
@@ -121,7 +155,44 @@ class CSP_API ResultBase
 	float ResponseProgress = 0.0f;
 
 	csp::common::String ResponseBody;
-	int FailureReason;
+	EResultBaseFailureReason FailureReason;
+
+private:
+	csp::common::Map<csp::common::String, EResultBaseFailureReason> XErrorCodeToFailureReason = {
```

Comment:
This map should be moved to WebService.cpp in an anonymous namespace, changes to a `std::map`, and be made static. We don't want to re-create this every time a new ResultBase instance is made. We should also only be mapping the quota errors and any error codes we already handle as that is all we need to cover in this ticket.